mmedit/models/base_models/base_edit_model.py

```diff
@@ -118,33 +118,32 @@ def forward(self,
         elif mode == 'loss':
             return self.forward_train(inputs, data_samples, **kwargs)
 
-    def convert_to_datasample(self, predictions: List[EditDataSample],
-                              data_samples: List[EditDataSample],
+    def convert_to_datasample(self, predictions: EditDataSample,
+                              data_samples: EditDataSample,
                               inputs: Optional[torch.Tensor]
                               ) -> List[EditDataSample]:
         """Add predictions and destructed inputs (if passed) to data samples.
 
         Args:
             predictions (List[EditDataSample]): The predictions of the model.
-            data_samples (List[EditDataSample]): The data samples loaded from
+            data_samples (EditDataSample): The data samples loaded from
                 dataloader.
             inputs (Optional[torch.Tensor]): The input of model. Defaults to
                 None.
 
         Returns:
             List[EditDataSample]: Modified data samples.
         """
-        for data_sample, pred in zip(data_samples, predictions):
-            data_sample.output = pred
 
+        # data_samples.output = predictions.pred_img
         if inputs is not None:
-            assert inputs.shape[0] == len(predictions), (
-                'The length of inputs and outputs must be same.')
-            for idx, data_sample in enumerate(data_samples):
-                # destruct input
-                destructed_input = self.data_preprocessor.destruct(
-                    inputs[idx], data_sample, 'img')
-                data_sample.set_data({'input': destructed_input})
+            destructed_input = self.data_preprocessor.destruct(
+                inputs, data_samples, 'img')
+            data_samples.set_tensor_data({'input': destructed_input})
+        data_samples = data_samples.split()
+
+        for data_sample, pred in zip(data_samples, predictions):
```

Comment:
The direct output (i.e., prediction) should be a tensor too. So, we can add prediction to data_samples first, and then slit them into a list of datasamples.